File: C++/arithmetic-subarrays.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        int n = nums.size();
        int m = l.size();
        vector<bool> ans(m);
        for (int i = 0; i < m; i++) {
            vector<int> tmp;
            ans[i] = true;
            
            for (int j = l[i]; j <= r[i]; j++) {
                tmp.push_back(nums[j]);
            }

            sort(tmp.begin(), tmp.end());
            
            int diff = tmp[1] - tmp[0];

            for (int j = 0; j < tmp.size() - 1; j++) {
                if (tmp[j+1] - tmp[j] != diff) {
                    ans[i] = false;
                    break;
                }
            }

        }
        return ans;
    }
};

File: Kotlin/arithmetic-subarrays.kt
class Solution {
    fun checkArithmeticSubarrays(nums: IntArray, l: IntArray, r: IntArray): List<Boolean> {
        val len = nums.size
        val queries = l.size
        var answer = MutableList<Boolean>(0, {false})
        for (q in 0 until queries) {
            val subLen = r[q] - l[q] + 1
            var sub = IntArray(subLen) { i -> nums[i + l[q]] }
            sub.sort()
            val delta = sub[1] - sub[0]
            var valid = true
            for (i in 1 until subLen) {
                if (sub[i] - sub[i-1] != delta) {
                    valid = false
                    break
                }
            }
            answer.add(valid)
        }
        return answer.toList()
    }
}